Services/DataBaseService/Repositories/UserRepository.cs

```diff
@@ -26,13 +29,39 @@ public void CreateUser(User user)
 
         public void CreateUserCredential(UserCredential userCredential)
         {
-            dbContext.UsersCredentials.Add(mapper.MapToDbUserCredential(userCredential));
-            dbContext.SaveChanges();
+            var dbCredential = dbContext.UsersCredentials.FirstOrDefault(uc => uc.Email == userCredential.Email);
+
+            if (dbCredential is null)
+            {
+                dbContext.UsersCredentials.Add(mapper.MapToDbUserCredential(userCredential));
+                dbContext.SaveChanges();
+            }
+            else
+            {
+                throw new ForbiddenException("User not found");
+            }
+            
         }
 
         public UserCredential GetUserCredential(string email)
         {
             var dbCredential = dbContext.UsersCredentials.FirstOrDefault(uc => uc.Email == email);
+            if (dbCredential is null)
+            {
+                throw new ForbiddenException("User not found");               

```

Comment:
Is this is forbidden or bad request? It is user can't login, right?

Reply:
It is NotFound

Reply:
In the beginning, I put NotFound here, but I thought, that NotFound exception is issued in cases when the page is not found, and not some data not found.
But if NotFound is used correctly here, I can replace it)